contracts/protocol/contracts/ERC20/ScoutTokenERC20Implementation.sol

```diff
@@ -34,15 +26,24 @@ contract ScoutTokenERC20 is
         _;
     }
 
-    constructor(address _admin) ERC20("Scout Token", "$SCOUT") {
-        MemoryUtils._setAddress(MemoryUtils.ADMIN_SLOT, _admin);
+    constructor() ERC20("Scout Token", "$SCOUT") {}
+
+    function initialize() external onlyAdmin {
+        require(
+            MemoryUtils._getBool(MemoryUtils.INITIALIZED_SLOT) == false,
+            "Already initialized"
+        );
+        address _admin = MemoryUtils._getAddress(MemoryUtils.ADMIN_SLOT);
         _mint(_admin, SUPPLY);
-        StorageSlot
-            .getAddressSlot(SUPERCHAIN_BRIDGE_SLOT)
-            .value = DEFAULT_SUPERCHAIN_BRIDGE_ADDRESS;
-        StorageSlot
-            .getAddressSlot(L2_MESSENGER_SLOT)
-            .value = DEFAULT_L2_MESSENGER_ADDRESS;
+    }
```

Comment:
We need to remove any mechanism for minting additional tokens and only mint once.

We can't do this in the constructor of implementation, since the state needs to live inside the proxy.

This allows us to support this case.

initialize method is a pattern established in Circle USDC which has initializers on its implementations